### PR TITLE
sci-libs/silo: bump to 4.11 and EAPI 8

### PR DESCRIPTION
**Pull request by scardracs**
- [X] Fix build (was a missing virtual/szip RDEPEND).
- [X] EAPI 8.
- [X] Add missing browser support.
- [X] Fix QA warning.

Closes: https://bugs.gentoo.org/828693

Package-Manager: Portage-3.0.30, Repoman-3.0.3
Signed-off-by: Marco Scardovi <marco@scardovi.com>

### CODE REVIEW

**Comment by scardracs** [comment]
Leave here for reference. it's actually broken

**Comment by scardracs** [comment]
I have wrote to original maintainer to ask for broken test. If you want you can merge that and wait for maintainer's answer for next release

File: sci-libs/silo/silo-4.11.ebuild
Comment (jsmolic) [inline sci-libs/silo/silo-4.11.ebuild:15]: Drop static-libs please

File: sci-libs/silo/silo-4.11.ebuild
Comment (jsmolic) [inline sci-libs/silo/silo-4.11.ebuild:28]: Are you sure we don't need these patches anymore?

File: sci-libs/silo/silo-4.11.ebuild
Comment (jsmolic) [inline sci-libs/silo/silo-4.11.ebuild:30]: Move below `SRC_URI` (to follow skel.ebuild)

**Comment by jsmolic** [comment]
> I have wrote to original maintainer to ask for broken test. If you want you can merge that and wait for maintainer's answer for next release

Yeah, that's okay, ideally we'd have an open upstream bug and link to it in the ebuild

File: sci-libs/silo/silo-4.11.ebuild
Comment (scardracs) [reply sci-libs/silo/silo-4.11.ebuild:28]: No, I'm not, but actually I need to repatch them as they all fails to apply

**Comment by jsmolic** [comment]
@scardracs can you drop the old version in a separate commit? I'm also getting unrecognized configure options
```
 * QA Notice: Unrecognized configure options:
 *
 *      --with-Qt-lib-dir
 *      --with-Qt-include-dir
 *      --with-Qt-bin-dir
 *      --with-Qt-lib-dir
 *      --with-Qt-include-dir
 *      --with-Qt-bin-dir
 * 
 ```

Also, I don't think tests are being ran at all
```
NOTICE: Silo test cases are not compiled by default.  To compile and test
NOTICE: them run "make check" from the silo "tests" directory. Otherwise
NOTICE: you can compile them individually with "make TESTNAME".  The test
NOTICE: names are: array, dir, extface, multi_test, partial_io, point, quad,
NOTICE: simple, ucd, ucdsamp3, testall, obj, onehex, oneprism, onepyramid,
NOTICE: onetet, subhex, TestReadMask, twohex, multispec, misc, sami,
NOTICE: newsami, specmix, spec, cpz1plt, group_test, listtypes, alltypes,
NOTICE: wave, multi_file, polyzl, csg, rocket, mmadjacency, largefile,
NOTICE: version, namescheme, efcentering, mk_nasf_pdb, ioperf, arbpoly2d,
NOTICE: readstuff, mat3d_3across, merge_block, test_mat_compression,
NOTICE: bcastopen, memfile_simple, empty, majorder, realloc_obj_and_opts,
NOTICE: testpdb, pdbtst, compression, grab, mk_nasf_h5, testhdf5, arrayf77,
NOTICE: arrayf90, curvef77, matf77, pointf77, quadf77, ucdf77, testallf77,
NOTICE: csgmesh, qmeshmat2df77, but "testall" is a good one to start with.
```

File: sci-libs/silo/silo-4.11.ebuild
Comment (jsmolic) [inline sci-libs/silo/silo-4.11.ebuild:14]: Swap SLOT and LICENSE order (per skel.ebuild)

File: sci-libs/silo/silo-4.11.ebuild
Comment (jsmolic) [inline sci-libs/silo/silo-4.11.ebuild:16]: From what I understand qt5 can't be disabled, so this flag should just be dropped if we need it unconditionally

File: sci-libs/silo/silo-4.11.ebuild
Comment (jsmolic) [inline sci-libs/silo/silo-4.11.ebuild:19]: Ditto

File: sci-libs/silo/silo-4.11.ebuild
Comment (jsmolic) [inline sci-libs/silo/silo-4.11.ebuild:23]: Sort alphabetically please

File: sci-libs/silo/silo-4.11.ebuild
Comment (jsmolic) [inline sci-libs/silo/silo-4.11.ebuild:30]: Small nit, I prefer `"${FILESDIR}"/${P}-autoreconf.patch` (with only FILESDIR in quotes), but no need to change if you don't want to

File: sci-libs/silo/silo-4.11.ebuild
Comment (jsmolic) [inline sci-libs/silo/silo-4.11.ebuild:26]: Drop newline

File: sci-libs/silo/silo-4.11.ebuild
Comment (jsmolic) [inline sci-libs/silo/silo-4.11.ebuild:48]: Use `emake` please, also you could just do `emake -C tests check` ;-)

File: sci-libs/silo/silo-4.11.ebuild
Comment (jsmolic) [inline sci-libs/silo/silo-4.11.ebuild:16]: Also, we don't need to add `test` to USE flags if we're not having any test-only dependencies

File: sci-libs/silo/silo-4.11.ebuild
Comment (scardracs) [reply sci-libs/silo/silo-4.11.ebuild:19]: Ditto?

File: sci-libs/silo/silo-4.11.ebuild
Comment (scardracs) [reply sci-libs/silo/silo-4.11.ebuild:16]: Test IUSE is required for conditional test

Comment (scardracs) [reply sci-libs/silo/silo-4.11.ebuild:16]: Required by Silex REQUIRED_USE

Comment (jsmolic) [reply sci-libs/silo/silo-4.11.ebuild:16]: But where is the test conditional, I don't see the `test` USE flag being used anywhere in the ebuild?

File: sci-libs/silo/silo-4.11.ebuild
Comment (jsmolic) [reply sci-libs/silo/silo-4.11.ebuild:19]: "Ditto" as this needs dropping too ;-)

File: sci-libs/silo/silo-4.11.ebuild
Comment (jsmolic) [reply sci-libs/silo/silo-4.11.ebuild:16]: Well it was previously, but now apparently `qt5` is always enabled so this requirement will always be satisfied

Comment (scardracs) [reply sci-libs/silo/silo-4.11.ebuild:16]: If I remove it and run ebuild ... test it warns me that test is not on IUSE

File: sci-libs/silo/silo-4.11.ebuild
Comment (jsmolic) [inline sci-libs/silo/silo-4.11.ebuild:17]: Can you add a bug reference here?

Comment (scardracs) [reply sci-libs/silo/silo-4.11.ebuild:17]: I'll add when I'm at home

Comment (jsmolic) [reply sci-libs/silo/silo-4.11.ebuild:17]: Sure, but write it in the ebuild ;)

**Comment by scardracs** [comment]
@jsmolic ready for merge

File: sci-libs/silo/files/silo-4.10.2-mpiposix.patch
Comment (jsmolic) [inline sci-libs/silo/files/silo-4.10.2-mpiposix.patch:1]: Dropping these patches should have been done in the second commit since we should have a working package at every commit, this way if you checkout right after the first commit you're left with version 4.10.2 in the tree but no patches it requires :(

Comment (scardracs) [reply sci-libs/silo/files/silo-4.10.2-mpiposix.patch:1]: I thought you merge all commits in 1 time

Comment (jsmolic) [reply sci-libs/silo/files/silo-4.10.2-mpiposix.patch:1]: Well yes, but the idea is we should preserve a working and functional tree at every commit. Suppose if you a new package version, but it doesn't compile, then you fix it up in another commit, the first commit isn't really complete since the package is broken, right? ;)

Comment (scardracs) [reply sci-libs/silo/files/silo-4.10.2-mpiposix.patch:1]: Right, I will take more consideration about that in future

Comment (jsmolic) [reply sci-libs/silo/files/silo-4.10.2-mpiposix.patch:1]: ...and it could also cause inconvenience if we for any reason need to revert a commit. Anyway this is just to keep in mind as a good practice when working with git, I think I'm going to merge this as is today

**Review by jsmolic — APPROVED** [review]
Thanks for your patience, hope I'm not being too nitpicky ;)

**Comment by jsmolic** [comment]
@scardracs Something is off with your sign-off, did you add it manually?
It should be `Signed-off-by: Marco Scardovi <marco@scardovi.com>` (note `<>` around the mail which you don't have currently). You can add it automatically with `-s` flag when using git (i.e. `git commit -s -m "..."`). I'll take care of this now, but watch out for this for future PRs

**Pull request closed without merging**
